packages/kbn-interpreter/src/public/batched_fetch.js

```diff
@@ -72,8 +73,8 @@ function createFuture() {
   let reject;
 
   return {
-    resolve(val) { return resolve(val); },
-    reject(val) { return reject(val); },
+    resolve: once((val) => resolve(val)),
```

Comment:
What's the value of `once` here? Promises can only resolve or reject once anyway.

Reply:
Hm. I thought I'd commented it. I *should* comment it. But there's an edge case where we might call resolve / reject more than once. Let's say we process the first item in the batch, then the network fails, so the over all batch operation rejects. That puts us into the catch on line 106 (I think) and that catch goes and tries to fail all of the promises.

Another way to do this (and maybe better / clearer?) would be to track "pending promises" and only reject those in the catch.